pandas/tests/series/test_constructors.py

```diff
@@ -1860,23 +1867,37 @@ class A(OrderedDict):
         series = Series(A(data))
         tm.assert_series_equal(series, expected)
 
-    def test_constructor_dict_multiindex(self):
-        d = {("a", "a"): 0.0, ("b", "a"): 1.0, ("b", "c"): 2.0}
-        _d = sorted(d.items())
-        result = Series(d)
-        expected = Series(
-            [x[1] for x in _d], index=MultiIndex.from_tuples([x[0] for x in _d])
-        )
-        tm.assert_series_equal(result, expected)
+    @pytest.mark.parametrize(
+        "data, expected_index_multi, expected_index_single",
```

Comment:
I don't believe we need two Boolean flags; can you remove one and use `if... else...`.